### PR TITLE
New multisig rewrite tests

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

**Review by alyn509 — APPROVED** [review]

File: contracts/multisig-improved/src/action_types/propose.rs [hunk 1]
```diff
@@ -73,11 +73,16 @@ pub trait ProposeModule:
         let own_shard = self.blockchain().get_shard_of_address(&own_sc_address);
         match action {
             Action::SendTransferExecuteEgld(call_data) => {
-                let other_sc_shard = self.blockchain().get_shard_of_address(&call_data.to);
                 require!(
                     call_data.egld_amount > 0 || !call_data.endpoint_name.is_empty(),
                     "proposed action has no effect"
                 );
+
+                if cfg!(debug_assertions) {
```
Comment (psorinionut) [inline contracts/multisig-improved/src/action_types/propose.rs:81]: What does this do? At a SC level?

Comment (andreiblt1304) [reply contracts/multisig-improved/src/action_types/propose.rs:81]: https://doc.rust-lang.org/rust-by-example/attribute/cfg.html

The `cfg!` macro allows for checks at runtime

**Review by andreiblt1304 — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]

**Review by ccorcoveanu — CHANGES_REQUESTED** [review]
This should go into the feat branch, when was this multisig merged into main? :)
I would like to have it audited before somebody using it in prod

**Comment by dorin-iancu** [comment]
Moved to new repo here: https://github.com/multiversx/mx-multisig-and-modules

**Pull request closed without merging**